File: public/usage-examples/graphics/draw_rectangle_on_window/draw-rectangle-on-window.py
import splashkit

window_width = 800
window_height = 600
rows = 8
cols = 8
rect_size = 60
offset_x = (window_width - cols * rect_size) / 2
offset_y = (window_height - rows * rect_size) / 2

my_window = splashkit.open_window("Checkerboard", window_width, window_height)

while not splashkit.window_close_requested(my_window):
    splashkit.clear_window(my_window, splashkit.color_white())
    
    for r in range(rows):
        for c in range(cols):
            x = offset_x + c * rect_size
            y = offset_y + r * rect_size
            fill_color = splashkit.color_black() if (r + c) % 2 == 0 else splashkit.color_light_gray()
            
            splashkit.fill_rectangle_on_window(my_window, fill_color, x, y, rect_size, rect_size)
            splashkit.draw_rectangle_on_window(my_window, splashkit.color_blue(), x, y, rect_size, rect_size)
    
    splashkit.refresh_window(my_window)

splashkit.close_window(my_window)

File: public/usage-examples/graphics/draw_rectangle_on_window/draw-rectangle-on-window.cpp
#include "splashkit.h"

int main()
{
    const int window_width = 800, window_height = 600, rows = 8, cols = 8, rect_size = 60;
    float offset_x = (window_width - cols * rect_size) / 2.0, offset_y = (window_height - rows * rect_size) / 2.0;
    window my_window = open_window("Checkerboard", window_width, window_height);

    while (!window_close_requested(my_window))
    {
        clear_window(my_window, COLOR_WHITE);
        for (int r = 0; r < rows; r++)
            for (int c = 0; c < cols; c++)
            {
                float x = offset_x + c * rect_size, y = offset_y + r * rect_size;
                color fill = ((r + c) % 2 == 0) ? COLOR_BLACK : COLOR_LIGHT_GRAY;
                draw_rectangle_on_window(my_window, fill, x, y, rect_size, rect_size);
                draw_rectangle_on_window(my_window, COLOR_BLUE, x, y, rect_size, rect_size);
            }
        refresh_window(my_window);
    }
    close_window(my_window);
    return 0;
}
